### PR TITLE
TanStack Integration

### PR DESCRIPTION
**Pull request by yanevdw**
This PR reflects the switch from RTK Query to TanStack Query.

This includes:
- Removing Redux and Redux Toolkit and all integration and implementation.
- Adding TanStack and Axios and implemented API endpoints and added interfaces as required.

### CODE REVIEW

File: src/services/api.ts [hunk 1]
```diff
@@ -1,41 +1,65 @@
...
+  const response = await axios.get(
+    "https://api.comick.fun/top?type=trending&comic_types=manwha&accept_mature_content=false"
+  );
+  return response.data.rank;
+}
+
+// Call the API to get the top/trending manhua.
+export async function fetchTopManhua(): Promise<RankDetails[]> {
+  const response = await axios.get(
+    "https://api.comick.fun/top?type=trending&comic_types=manwha&accept_mature_content=false"
```
Comment (Hugovs2000) [inline src/services/api.ts]: I think you meant Manhau so this: `"https://api.comick.fun/top?type=trending&comic_types=manhua&accept_mature_content=false"`

**Review by Hugovs2000 — CHANGES_REQUESTED** [review]
Looks good, just the one comment on the Url.

**Review by Julianna-Venter — APPROVED** [review]
Looks good to me

**Review by Hugovs2000 — APPROVED** [review]
All good 👍🏻

**Pull request merged**